contracts/InterchainTokenFactory.sol

```diff
@@ -283,8 +283,9 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
             if (bytes(originalChain).length == 0) {
                 chainNameHash_ = chainNameHash;
             } else {
-                chainNameHash_ = keccak256(bytes(originalChain));
+                revert NotSupported();
```

Comment:
an overloaded version of these methods should be created without the `originalChain` arg